### ISSUE TITLE
Using an Azure storage block for deployment build throws unexpected keyword argument ignore_file error

### ISSUE THREAD

**Issue by scoussens** [PrefectHQ/prefect#6204]
### Describe the bug

When running the following deployment command:

`prefect deployment build ./custom_flow.py:my_docker_flow --name docker-custom -ib docker-container/docker-prefect-1-0-0 -t test -sb azure/azure-storage-block`

The command throws the following error:

`TypeError: put_directory() got an unexpected keyword argument 'ignore_file'`

Prefect 2.0.2

### To Reproduce
Flow Code:
```python
from prefect import flow, get_run_logger
import os

@flow
def my_docker_flow():
    logger = get_run_logger()
    files = os.listdir()
    logger.info(files)


if __name__ == "__main__":
    my_docker_flow()
```

Using Prefect 2.0 cloud.
Created an Azure storage block via the UI.
Testing and worked with Remote Files storage block, and the only change was the storage block in the deployment command.

```
prefect deployment build ./custom_flow.py:my_docker_flow --name docker-custom -ib docker-container/docker-prefect-1-0-0 -t test -sb azure/azure-storage-block
```

### Error

```
Found flow 'my-docker-flow'
Manifest created at '/home/-/prefect/custom/my_docker_flow-manifest.json'.
Traceback (most recent call last):
  File "/home/-/.local/lib/python3.9/site-packages/prefect/cli/_utilities.py", line 41, in wrapper
    return fn(*args, **kwargs)
  File "/home/-/.local/lib/python3.9/site-packages/prefect/utilities/asyncutils.py", line 193, in wrapper
    return run_async_in_new_loop(async_fn, *args, **kwargs)
  File "/home/-/.local/lib/python3.9/site-packages/prefect/utilities/asyncutils.py", line 140, in run_async_in_new_loop
    return anyio.run(partial(__fn, *args, **kwargs))
  File "/home/-/.local/lib/python3.9/site-packages/anyio/_core/_eventloop.py", line 70, in run
    return asynclib.run(func, *args, **backend_options)
  File "/home/-/.local/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 292, in run
    return native_run(wrapper(), debug=debug)
  File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/home/-/.local/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 287, in wrapper
    return await func(*args)
  File "/home/-/.local/lib/python3.9/site-packages/prefect/cli/deployment.py", line 470, in build
    file_count = await storage.put_directory(ignore_file=".prefectignore")
TypeError: put_directory() got an unexpected keyword argument 'ignore_file'
``` 

### Versions

- OS: Ubuntu 20.04
- Python: 3.9
- Prefect: 2.0.2

**Comment by madkinsz** [comment]
Fixed by #6195

**Comment by Trymzet** [comment]
Upgrading to latest Prefect version fixed this for me.
